### DISCUSSION TITLE
GoPro 12 - !MBOOT not working

### DISCUSSION THREAD

**Discussion by Josef1406** [gopro/labs#729] [Q&A]
My GoPro 12 runs without a battery, but I use an external power supply (with external Timerrelais).

The GoPro should boot automatically in the event of a power failure, then she should turn on the recording.

The GoPro is supposed to upload at 10:00 a.m, then the camera should be switched off.

If I then switch the power off/on (external timer), boot the camera again, start recording and upload again at 10:00 a.m.

My problem is that:
If i the external power switching OFF/ON, the camera doesn't boot automatically, so the script doesn't start either.

My script looks like this:

![Upload_on_10_00](https://github.com/gopro/labs/assets/137929364/7afa8c0e-5866-43f7-9bec-08dbc87382a7)
![Upload_on_10_00-1_21_2024, 20_13_09](https://github.com/gopro/labs/assets/137929364/844e25df-7fad-4fe4-9c99-f13271af950b)

**Comment by dnewman-gpsw** [comment]
Use !MWAKE=2 not 1.  2 is boot on power, 1 is only boot on power if there was a wake event scheduled

**Comment by dnewman-gpsw** [comment]
What are you using NAKEd emulation for?

**Reply by Josef1406** [reply]
ok i test it,
!MNAKE=1 - i doesnt need ?

**Reply by dnewman-gpsw** [reply]
It does have effects, but this project shouldn't need it.

**Reply by Josef1406** [reply]
Now it Start on Power on :-) thanks for your help

What ist the command !MNAKE=2  ?

**Reply by Josef1406** [reply]
If i scan a new QR-Script, have i before complete RESET my GoPro, or delete the SD-Card or scan the QR !RESET ?
What is the best case ?

**Reply by Josef1406** [reply]
This is my idea, for a long time "Timelaps". i think, so i can controll every day for a new record in the cloud. And i think so is no problem, if a short powerfail.
What do you meen, is this a good solution?
[Timelaps.pdf](https://github.com/gopro/labs/files/14002790/Timelaps.pdf)

**Reply by Josef1406** [reply]
![Timelaps](https://github.com/gopro/labs/assets/137929364/a5530366-62e7-4b3e-bebd-5842a1c2efc3)

**Reply by dnewman-gpsw** [reply]
!MNAKE=1 emulates a naked camera, used for FPV applications. Sometimes the side effects are benefits other applications.  

!Mwxyz commands are permanent.  They can only be removed by setting them to zero or !RESET, as they are not saved to the SD card.

**Reply by dnewman-gpsw** [reply]
With HERO12, you can do all this reliably without the external timer.  Earlier models the external timer can help.

**Reply by Josef1406** [reply]
Do you have a suggestion on how I should design the QR code?

The GoPro should do a timelapse (16:9|4K|30s|L) from 6:00 a.m. to 10:00 p.m. and transfer it to the cloud after 10:00 p.m.
after that, the camera could sleep until 6:00.
Attention: if AC power failure (could happen without a battery) the script should start normally again.

Do you have a suggestion for me?

I would be very grateful to you

**Reply by dnewman-gpsw** [reply]
You should have the battery installed and fully charged, with either a Voltaic V50 battery to AC USB power external. The internal battery helps preserve the internal clock time and date.  Set the video mode in the default timelapse video present, don't set this in the script on every boot.  So to set the default timelapse the preset use:  mTp.1800r4fL0dT

Then a basic script would be something like, starts tomorrow at 6am:  !06:00N!S!57600E!2U!R

more complex, starts today, anytime after 6am:  ```<r0>06:00!S>r0>22:00!E+!2U+!06:00N!R```

**Reply by Josef1406** [reply]
fine :-)
Ok, the !MWake=2 command i need only if no akku and Power OFF/ON - i understand.
With akku(and USB Charger), i press the PowerButton on GoPro and start with the QR-command "06:00!S>r0>22:00!E+!2U+!06:00N!R".
Can i restart the script exterally ?
The camera is hanging high up on a pole.

**Reply by Josef1406** [reply]
But I understand the logic:

- I can see whether the script runs after the start after every upload at 10:00 p.m. in the cloud. 
- If there is no upload at 10 p.m.in the cloud (i can check it at home), it means there is something wrong with the GoPro. 
- Then I have to go there, check the problem and possibly restart with the QR code (06:00!S>r0>22:00!E+!2U+!06:00N!R) again.

Did I understand that correctly ?

Thanks
Josef

**Reply by Josef1406** [reply]
I still have 1 question:

Is an upload only possible after the recording has ended? 
It would be cool to upload at 11:00 am and 10:00 pm. 
The recording should continue to run during the upload so that I don't lose any information.

Thanks

**Reply by Josef1406** [reply]
i have test your example:  _**more complex, starts today, anytime after 6am: 06:00!S>r0>22:00!E+!2U+!06:00N!R -**_-> the Camera says "Code non GOPRO".
If i added an "!" ->  now the QR is "!06:00!S>r0>22:00!E+!2U+!06:00N!R" -> it works.

But the GoPro says "StartTime at  06:00 Chancel ...." 
Recording does not start immediately.

**Reply by dnewman-gpsw** [reply]
There is a bug in the forum so that it doesn't show all of the command, I have to put it code a code segment to reveal the full command ```<r0>06:00!S>r0>22:00!E+!2U+!06:00N!R```

**Reply by dnewman-gpsw** [reply]
The upload can only run while not recording.

**Comment by Josef1406** [comment]
i see this:
![image](https://github.com/gopro/labs/assets/137929364/648b4210-99a1-4864-a3e5-92e3abbe57d3)
is that so ok ?

**Reply by dnewman-gpsw** [reply]
Not that is embedded in a code segment it is all there.  Things between < and > can disappear, I just forgot.

**Comment by Josef1406** [comment]
OK:-)

**I now have the following workflow:**

- I use the GoPro with a battery and an AC charger
- I turn on the GoPro
- I scan the QR command “!RESET”
- I connect the GoPro to the WiFi

- I mount the GoPro and set the correct field of view

**Now I scan the QR command "!06:00NmTfLp.30!S!57600E!U!1R"**

_The Gopro Start on next Day at 6:00 with Recording, Stop at 22:00 and upload the Video.
Then the GoPro sleeps until 6:00_

What do you think, is this solution good?

**Reply by dnewman-gpsw** [reply]
Sure. I'm not sure why you need the !RESET, that is only to fix screw-ups you have forgotten about.

**Reply by Josef1406** [reply]
Ok I didn't use !RESET.

Everything is working - almost

Can you connect to a WiFi hotspot with QR-Comand?
So select SSID, enter password.

I have a different one at the customer's than at home.

**Reply by dnewman-gpsw** [reply]
Just use Quik to connect to their network.  Do a test upload at the location.
